packaging/aznfs/RPM/aznfs.spec

```diff
@@ -145,6 +148,28 @@ if [ $1 == 0 ]; then
 		fi
 	fi
 
+	existing_mounts_for_nfsv4FilesShares=$(cat /opt/microsoft/aznfs/data/aznfs_files_mountmap 2>/dev/null | egrep '^\S+' | wc -l)
+	if [ $existing_mounts_for_nfsv4FilesShares -ne 0 ]; then
+		echo
+		echo -e "${RED}There are existing Azure Files NFS mounts using aznfs mount helper, they will not be tracked!" > /dev/tty
+		echo -n -e "Are you sure you want to continue? [y/N]${NORMAL} " > /dev/tty
+		read -n 1 result < /dev/tty
+		echo
+		if [ "$result" != "y" -a "$result" != "Y" ]; then
+			echo "Removal aborted!"
+			if [ "DISTRO" != "suse" -a ! -f /etc/centos-release ]; then
+				echo
+				echo "*******************************************************************"
+				echo "Unfortunately some of the anzfs dependencies may have been uninstalled."
+				echo "aznfs mounts may be affected and new aznfs shares cannot be mounted."
+				echo "To fix this, run the below command to install dependencies:"
+				echo "INSTALL_CMD install conntrack-tools iptables bind-utils iproute util-linux nfs-utils NETCAT_PACKAGE_NAME stunnel"
+				echo "*******************************************************************"
+				echo
+			fi
+			exit 1
```

Comment:
I think the v3 and v4 check should all happens unconditionally.
Basically we need to check both v3 and v4, we should not bypass v4 if v3 already detect. Because they are OK to aborted for their v3 mount, doesn't mean they are OK to aborted for their v4 mount. 
Unless we are sure v3 and v4 will never mount in the same VM.

Reply:
Talked to Rena. Customers can mount both v3 and v4 shares on the same vm. So merged code for v3 and v4 here.